redis/src/aio/connection.rs

```diff
@@ -502,42 +489,34 @@ impl Stream for SplitPubSubStream {
         } = self.project();
 
         if let Some(msg) = queue.pop_back() {
-            return Poll::Ready(Some(Ok(msg)));
+            return Poll::Ready(Some(msg));
         }
 
         loop {
             if resp_waiting.is_some() {
                 match resp_tx.poll_ready(cx) {
                     Poll::Ready(Ok(_)) => {
                         if let Some(resp) = resp_waiting.take() {
-                            map_response_forward_err(resp_tx.start_send(resp))?;
-                        }
-                    }
-                    Poll::Ready(Err(err)) => {
-                        if !err.is_disconnected() {
-                            map_response_forward_err(Err(err))?;
-                        } else {
-                            drop(resp_waiting.take());
+                            drop(resp_tx.start_send(resp));
                         }
                     }
+                    Poll::Ready(Err(_)) => drop(resp_waiting.take()),
                     Poll::Pending => return Poll::Pending,
                 };
             }
 
             match stream.as_mut().poll_next(cx) {
                 Poll::Ready(Some(Ok(Ok(value)))) => match Msg::from_value(&value) {
-                    Some(msg) => return Poll::Ready(Some(Ok(msg))),
+                    Some(msg) => return Poll::Ready(Some(msg)),
                     _ => {
                         // This should only be reachabled if the resp_waiting conditional above is
                         // false.
                         *resp_waiting = Some(value)
                     }
                 },
-                Poll::Ready(Some(Err(err)) | Some(Ok(Err(err)))) => {
-                    return Poll::Ready(Some(Err(err)));
-                }
-                Poll::Ready(None) => return Poll::Ready(None),
+                Poll::Ready(Some(Err(_)) | Some(Ok(Err(_)))) => {}
```

Comment:
making sure - will the stream terminate once the pubsub is disconnected from the server? can we add tests that drop the server and see that the stream completes?